torch/nn/functional.py

```diff
@@ -2486,6 +2499,17 @@ def _check_size_scale_factor(dim):
                 and len(scale_factor) != dim:
             raise ValueError('scale_factor shape must match input shape. '
                              'Input is {}D, scale_factor size is {}'.format(dim, len(scale_factor)))
+        if scale_factor is not None:
+            if use_scale_factor is None:
+                is_float_scale_factor = any(not float(scale).is_integer() for scale in _ntuple(dim)(scale_factor))
```

Comment:
The `float(scale).is_integer()` seems suspicious to me. Why are we not just checking if `scale` is an integer instead of casting it to a float and then checking? I think at least a comment should be added.

Reply:
scale.is_integer() would fail if scale is not a float. 
I could have checked 'if scale is int or if scale is float and scale.is_integer()', which seemed to me more complex than a simple call 'float(scale).is_integer()'

Reply:
Why are we warning if scale_factor contains `1.0` compared to `1.1`, or if it contains integers? as far as i can tell the function still works regardless:
`[int(math.floor(float(input.size(i + 2)) * scale_factors[i])) for i in range(dim)]`
and
`[int(math.floor(float(input.size(i + 2)) * scale_factors[i])) for i in range(dim)]`

Reply:
We are warning if scale_factor contains something like 1.1 but not 1.0.
This was suggested by @fmassa to avoid always printing the warning even when the change will not affect the users.

The output of interpolate will only differ when the scale_factors are floats as explained here https://github.com/pytorch/pytorch/pull/28324#discussion_r355661922.